src/ucp/proto/proto_common.c

```diff
@@ -661,9 +660,19 @@ void ucp_proto_request_zcopy_completion(uct_completion_t *self)
 
 int ucp_proto_is_short_supported(const ucp_proto_select_param_t *select_param)
 {
-    /* Short protocol requires contig/host */
-    return (select_param->dt_class == UCP_DATATYPE_CONTIG) &&
-           UCP_MEM_IS_HOST(select_param->mem_type);
+    return (select_param->dt_class == UCP_DATATYPE_CONTIG);
+}
+
+int ucp_proto_mem_is_accessible(const ucp_proto_select_param_t *select_param,
+                                uct_ep_operation_t memtype_op)
+{
+    if (UCP_MEM_IS_ACCESSIBLE_FROM_CPU(select_param->mem_type)) {
+        return 1;
+    } else if (memtype_op != UCT_EP_OP_LAST) {
```

Comment:
could short protocol be selected for 0-byte GPU memory send?

Reply:
Yes, it can